src/gui/src/drcWidget.cpp

```diff
@@ -0,0 +1,538 @@
+    }
+  }
+  return selections;
+}
+
+void DRCWidget::loadReport(const QString& filename)
+{
+  violations_.clear();
+
+  if (filename.endsWith(".rpt")) {
+    loadTRReport(filename);
+  } else if (filename.endsWith(".ascii")) {
+    loadASCIIReport(filename);
+  } else {
+    logger_->error(utl::GUI, 32, "Unable to determine type of {}", filename.toStdString());
+  }
+
+  updateModel();
+}
+
+void DRCWidget::loadTRReport(const QString& filename)
+{
+  std::ifstream report(filename.toStdString());
+  if (!report.is_open()) {
+    logger_->error(utl::GUI, 30, "Unable to open TritonRoute DRC report: {}", filename.toStdString());
+  }
+
+  std::regex violation_type("\\s*violation type: (.*)");
+  std::regex srcs("\\s*srcs: (.*)");
+  std::regex bbox_layer("\\s*bbox = (.*) on Layer (.*)");
+  std::regex bbox_corners("\\s*\\(\\s*(.*),\\s*(.*)\\s*\\)\\s*-\\s*\\(\\s*(.*),\\s*(.*)\\s*\\)");
+
+  auto tech = block_->getDataBase()->getTech();
+  while (!report.eof()) {
+    std::string line;
+    std::smatch base_match;
+
+    // type of violation
+    std::getline(report, line);
+    std::string type;
+    if (std::regex_match(line, base_match, violation_type)) {
+      type = base_match[1].str();
+    } else {
+      continue;
+    }
+
+    // sources of violation
+    std::getline(report, line);
+    std::string sources;
+    if (std::regex_match(line, base_match, srcs)) {
+      sources = base_match[1].str();
+    } else {
+      continue;
+    }
+
+    std::getline(report, line);
+    // bounding box and layer
+    if (std::regex_match(line, base_match, bbox_layer)) {
+      std::string bbox = base_match[1].str();
+      odb::dbTechLayer* layer = tech->findLayer(base_match[2].str().c_str());
+
+      odb::Rect rect;
+      if (std::regex_match(bbox, base_match, bbox_corners)) {
+        rect.set_xlo(std::stod(base_match[1].str()) * block_->getDbUnitsPerMicron());
+        rect.set_ylo(std::stod(base_match[2].str()) * block_->getDbUnitsPerMicron());
+        rect.set_xhi(std::stod(base_match[3].str()) * block_->getDbUnitsPerMicron());
+        rect.set_yhi(std::stod(base_match[4].str()) * block_->getDbUnitsPerMicron());
+      }
+
+      std::vector<std::any> srcs_list;
+      std::stringstream srcs_stream(sources);
+      std::string single_source;
+
+      // split sources list
+      while(getline(srcs_stream, single_source, ' ')) {
+        if (single_source.empty()) {
+          continue;
+        }
+
+        auto pin_loc = single_source.find("PIN/");
+        if (pin_loc != std::string::npos) {
+          std::string pin_name = single_source.substr(pin_loc + 1);
+          if (pin_name != "OBS") {
+            // bterm
+            odb::dbBTerm* bterm = block_->findBTerm(pin_name.c_str());
+            if (bterm != nullptr) {
+              srcs_list.push_back(bterm);
+            }
+          }
+        } else {
+          auto inst_loc = single_source.find("/");
+          if (inst_loc != std::string::npos) {
```

Comment:
This will work today but when we have net names with a / it will break.  Do we need to add a type prefix in the report?

Reply:
A type prefix would simplify this a lot... and make it easier to pick the right item. Once there is some hierarchy that will almost certainly be required otherwise this would be pretty tricky.

Reply:
You can modify FlexDR::reportDRC and coordinate it with your parser.